src/Language/PureScript/AST/Declarations.hs

```diff
@@ -83,7 +91,7 @@ data SimpleErrorMessage
   | OrphanTypeDeclaration Ident
   | RedefinedIdent Ident
   | OverlappingNamesInLet
-  | UnknownName (Qualified Name)
+  | UnknownName UnknownNameParams
```

Comment:
I think that when a particular sum type will always include a particular piece of information, it's often better to normalize it so that that field isn't repeated across different variants of the sum. In this case an `UnknownName` error will always include the name which was unknown, so I'd prefer to leave it as a `Qualified Name` and always have that as the first field of the constructor, rather than using `Either` and representing it as either a `Qualified Name` in the `Right` case or as a `Text` in the `Left` case . So I think I'd prefer:
```haskell
| UnknownName (Qualified Name) (Maybe (SourceType, Context))
```
and then the second field will only be present in the case of unknown name errors at the term level where we can use the type checker to provide some additional information. Do you think that would work, or am I missing something and was there a particular reason you went for `Either` instead?

Reply:
There wasn't a particular reason I went for `Either`.  I am changing it over now!